### ISSUE TITLE
[BETTER] Summarise which floor you're kicked on in a treasure dungeon

### ISSUE THREAD

**Issue by Jaksuhn** [NadyaNayme/TidyChat#68]
## Better Message Request

**What is the current in-game message? Please provide a screenshot or copy/paste of the text.**
There are two particular messages during the treasure dungeons. You would have to wait until the second message is triggered then parse the most recent "chamber opens" message for this to work.

"The gate to the [1st/2nd/3rd/4th/5th/6th] chamber opens."
"A trap is triggered! You are expelled from the area!"

**What would a better message look like?**
[TidyChat] Kicked out on [1st/2nd/3rd/4th/5th/6th] floor.

**Why is this an improvement?**
There's about half a dozen different system messages that appear frequently in treasure dungeons. and are not meaningful. I otherwise have system messages disabled entirely and have only recently started enabling them just while I'm in there so I can tell what floor I was kicked out on as I record this data in a spreadsheet. The current duty which tells what floor you're on disappears relatively quickly and once it's gone there's no way to tell outside of the last system message.

**Comment by NadyaNayme** [comment]
Treasure dungeons being... maps or the new dungeon? I imagine maps but I haven't actually done a map dungeon well... literally ever in all my years of playing. I sell my maps. ^^;

Going to be honest I haven't (actively) played in almost 6 months. I just try and make sure TidyChat gets updated for any Dalamud API bumps. I'm in the middle of planning a pretty major refactor to make the code less of a pain in the arse to add new filters for and to allow Debug mode to be more useful.

I'll try and figure out a good way to handle this - will probably be similar to how I handled BetterCommendations before Midori helped me sig it. In that I'll track any "chamber opens" messages and then hide the "a trap is triggered!" message and replace it with a BetterMessage. 

I 100% expect another `You received 1 commendation from completing dancing thancred was pretty pleasant when I ended up kugane o.` situation to happen but at least it'll be funny.

**Comment by Jaksuhn** [comment]
Yeah treasure dungeons just refer to all of the map dungeons (there's 8). I'll have to double check the messages are identical in the different variations, and whether or not the "chamber opens" appears for the initial floor, cause otherwise I'm not sure how that case would be handled unless you have a situation like assuming that message's absence means it's the first floor.

**Comment by Kagekazu** [comment]
Should be fixed in next update.
